lib/jekyll/document.rb

```diff
@@ -263,47 +272,61 @@ def read(opts = {})
         @data = SafeYAML.load_file(path)
       else
         begin
-          defaults = @site.frontmatter_defaults.all(relative_path, collection.label.to_sym)
-          merge_data!(defaults, source: "front matter defaults") unless defaults.empty?
-
-          self.content = File.read(path, Utils.merged_file_read_opts(site, opts))
-          if content =~ YAML_FRONT_MATTER_REGEXP
-            self.content = $POSTMATCH
-            data_file = SafeYAML.load(Regexp.last_match(1))
-            merge_data!(data_file, source: "YAML front matter") if data_file
-          end
-
+          merge_defaults
+          read_content(opts)
           post_read
         rescue SyntaxError => e
           Jekyll.logger.error "Error:", "YAML Exception reading #{path}: #{e.message}"
-        rescue Exception => e
-          if e.is_a? Jekyll::Errors::FatalException
-            raise e
-          end
+        rescue StandardError => e
```

Comment:
Please revert this and send a separate pull request for this, this has been brought up by me before but there is a larger surface that needs to be tested and thought about, @parkr has his reasons for this and this will block your pull because it's a feature change.